controller/notificationController.js

```diff
@@ -0,0 +1,193 @@
+const ResponseTypeEnum = require('../model/enums').ResponseTypeEnum;
+const User = require('../model/userModel');
+
+
+const create = (req, res) => {
+
+    const user = req.decoded;
+
+    if (!user) {
+        return res.status(HttpStatus.BAD_REQUEST).json({
+            message: 'Bad request.'
+        });
+    }
+
+    let notification = new Notification(req.body);
+    notification.companyID = user.companyID;
+
+    notification.save().then(newNotification => {
+        // send new notification only to those that have accepted the invite
+        // we could probably add some sort of message queue for those whose status is pending so that when they accept they 
+        // can receive all those notifications to that they don't miss anything
+        Invite.find({
+            eventID: newNotification.eventID,
+            responseType: ResponseTypeEnum['Accepted']
+        }, {
+            userID: 1 // only need the userID - cut down on network traffic
+        }).then(invites => {
+            const userIDs = invites.map(invite => invite.userID);
+            User.updateMany({
+                _id: {
+                    $in: userIDs
+                }
+            }, {
+                $push: {
+                    notifications: {
+                        notificationID: newNotification._id
+                    }
+                }
+            }).then(() => {
+                return res.status(HttpStatus.CREATED).json(newNotification);
+            }).catch(err => {
+                return res.status(HttpStatus.BAD_REQUEST).json({
+                    message: ValidationUtil.buildErrorMessage(err, 'create', 'notification')
+                });
+            });
+        }).catch(() => {
+            return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+                message: 'Internal server error.'
+            });
+        });
+    }).catch(err => {
+        return res.status(HttpStatus.BAD_REQUEST).json({
+            message: ValidationUtil.buildErrorMessage(err, 'create', 'notification')
+        });
+    });
+
+};
+
+const getAll = (req, res) => {
+
+    const user = req.decoded;
+    const read = req.params.read;
+
+    if (!user) {
+        return res.status(HttpStatus.BAD_REQUEST).json({
+            message: 'Bad request.'
+        });
+    }
+
+    const notificationIDs = user.notifications.filter(notification => notification.read.toString() == read).map(notification => notification.notificationID);
+
+    Notification.find({
+        _id: {
+            $in: notificationIDs
+        }
+    }).then(notifications => {
+        return res.status(HttpStatus.OK).json(notifications);
+    }).catch(() => {
+        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+            message: 'Internal server error'
+        });
+    });
+
+};
+
+// I don't think admins should be able to delete notifications
+// Maybe we should make them delete automatically after event is marked as finished, dunno => DISCUSS
```

Comment:
I also think that deleting the notifications should not be done by the admin because people should theoretically be notified that something that they read on the notification is not valid anymore, that makes the whole process more complex. If someting changes admin should just send another notification to explain the changes.